### PR TITLE
Sitemap Overhaul

### LINKED ISSUE #55: Sitemap is confusing
**Is your feature request related to a problem? Please describe.**
I feel like the way features are spread is pretty unintuitive right now; the main problem being that a lot of features are duplicated across many pages, making the user confused as to where they can find a feature.
In fact, most of the features in the Dashboard are duplicated from other pages, which means that, compounded with the fact that Dashboard is the default page, the user is disincentivised from exploring the site.

**Describe the solution you’d like**
A better way to look at this is to look at the items in Navigation as groups of features, where if a user wants to do something in a category, they can just select a group they think would have that action.
For example: if a teacher wants to know what's the next class they're going to teach, they would go to the Teach tab because they're going to teach the class.
Since features are easier to find on their own, we don't need Dashboard.

**Additional context**
I tried to implement this with this new sitemap (mostly existing pages rearranged to minimise the amount of work we need to do for this).
image

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
image
image
image

**Features**
- New sitemap
  Resolves #55
  - Removed `/s/` and `/t/` prefixes
  - Removed Student Dashboard and Teacher Dashboard
    - Adequately replaced by other features already introduced
  - Grouped pages together when logical
  - See full changes in Notes below
- Matches paths with RegExes instead of `startsWith` and `includes` methods in middleware
- New Landing backgrounds
- Page dim during page load
- Updates to error handling:
  - Client-side error fallback
  - New 404 page
  - 500 page
- Migrated to `LangCode` and `getLocaleString` when given opportunity

**Fixes**
- View Class icons are too small
- Schedule crashes if the user is not assigned a subject
- Edit article button in Article is not translated
- Chip Lists in Cards has a margin originally added as a workaround to a problem that now doesn’t exist

**Notes**
Data unavailability means some of the features migrated to the new sitemap are not tested.

The new sitemap is as follows:

- Landing
  - /
- Error handling
  - /404
  - /500
- About
  - /about
- Account
  - /account
  - /account/login
- Admin
  - /admin
  - /admin/classes
  - /admin/news
  - /admin/news/create/form
  - /admin/news/create/info
  - /admin/news/create/stats
  - /admin/news/edit/info/[infoID]
  - /admin/schedule/[grade]
  - /admin/students
  - /admin/subjects
  - /admin/teachers
- Class
  - /class/[classNumber]/manage
  - /class/[classNumber]/students
  - /class/[classNumber]/teachers
  - /class/[classNumber]/view
- Learn
  - /learn
  - /learn/[classNumber]
- News
  - /news
  - /news/form/[formID]
  - /news/info/[infoID]
  - /subject/[subjectID]
- Teach
  - /teach
  - /teach/schedule

### CODE REVIEW

File: src/components/Layout.tsx [hunk 1]
```diff
@@ -63,99 +63,79 @@ const Layout = ({
...
     const role = session?.user?.user_metadata.role;
-    const isAdmin = session?.user?.user_metadata.isAdmin;
 
     // Decide the Navigation the user is going to see based on their role
-    // Append the Admin Nav Item to the Navigation if the user is an admin
-    if (role == "student")
-      setNavItems([...studentNav, isAdmin ? adminNavItem : newsNavItem]);
-    else if (role == "teacher")
-      setNavItems([...teacherNav, isAdmin ? adminNavItem : newsNavItem]);
+    if (role == "student") setNavItems(studentNav);
```
Comment (Jimmy-Tempest) [inline src/components/Layout.tsx:137]: where did the admin part go?

File: src/components/error/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// External libraries
+import { Component, FunctionComponent, ReactNode } from "react";
+
+class ErrorBoundary extends Component<
```
Comment (Jimmy-Tempest) [inline src/components/error/ErrorBoundary.tsx:4]: UMM, class component

**Review by Jimmy-Tempest — APPROVED** [review]
from the code perspective LGTM!!!

File: src/components/error/ErrorBoundary.tsx (hunk 2 above)
Comment (SiravitPhokeed) [reply src/components/error/ErrorBoundary.tsx:4]: Umm, functional component is not supported here.  At least, it doesn’t look like it.
See [documentation](https://nextjs.org/docs/advanced-features/error-handling).

Let me know though if you found a functional component example.

File: src/components/Layout.tsx (hunk 1 above)
Comment (SiravitPhokeed) [reply src/components/Layout.tsx:137]: Seems like I wasn’t done working on Navigation when I made the PR. Oh well, things slip through the cracks.

Anywho, if you recall from the sitemap diagram I made a while ago, I was going to merge Me and Admin, and that’s why Admin shouldn’t be there. 

I’m thinking, though, if Admin should be back on Navigation. After all, an admin would probably go to Admin enough to warrant one. Thoughts?

**Review by Temp9699 — APPROVED** [review]
LGTM created new issue #71

**Comment by SiravitPhokeed** [comment]
As mentioned in #71, merging this to `main` right now would break the beta. That should be fixed first before merging.

**Pull request merged**